src/components/MainPage.js

```diff
@@ -51,60 +52,102 @@ const styles = () => ({
 
 type Props = {
   getMovies: Object,
-  data: Object,
+  fetchedDataBundle: Object,
   classes: Object,
   error: Object,
   dispatch: Dispatch,
+  location: Object,
 };
 
 type State = {
   inputValue: number | string,
-  favorites: Array<number>,
+  activePage: number,
+  favorites: string,
 };
 
 class MainPage extends Component<Props, State> {
   componentDidMount() {
-    this.props.getMovies.movies();
+    const urlParamsValues = queryString.parse(this.props.location.search, '?');
+    const defaultSortParam = 'popularity.desc';
+    this.setState({ activePage: urlParamsValues.page || 1 });
+    this.props.getMovies.movies(
+      null,
+      urlParamsValues.page || 1,
+      defaultSortParam
+    );
+    const favorites = localStorage.getItem('favorites');
+    if (!favorites) {
+      localStorage.setItem('favorites', JSON.stringify([]));
+    }
   }
 
   handleSearchChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
-    this.setState({ inputValue: event.target.value });
+    this.setState({
+      inputValue: event.target.value,
+      activePage: 1,
+    });
   };
 
   handleSubmit = (event: SyntheticInputEvent<HTMLInputElement>) => {
     event.preventDefault();
-    this.props.getMovies.movies(this.state.inputValue);
+    const urlParamsValues = queryString.parse(this.props.location.search, '?');
+    const defaultSortParam = 'popularity.desc';
+    this.setState({ activePage: urlParamsValues.page });
+    this.props.getMovies.movies(
+      this.state.inputValue,
+      urlParamsValues.page || this.state.activePage,
+      defaultSortParam
+    );
   };
 
-  //TODO: add functionality to handle list of favorites movies and bottom navigation
-  handleBottomNavigationChange = (
-    event: SyntheticInputEvent<HTMLInputElement>
-  ) => {
-    event.preventDefault();
-    console.log('handleBottomNavigationChange');
+  handlePageChange = (pageNumber: number) => {
+    const { inputValue = null } = this.state;
+    this.setState({ activePage: pageNumber });
+    this.props.getMovies.movies(inputValue, pageNumber);
+    const history = createBrowserHistory({ basename: '/' });
+    history.push(`?page=${pageNumber}`);
   };
 
-  openListOfFavorites = (event: SyntheticInputEvent<HTMLInputElement>) => {
-    event.preventDefault();
-    console.log('openListOfFavorites');
+  addMovieToListOfFavorites = (id: number) => () => {
+    const favorites = localStorage.getItem('favorites');
+    if (favorites) {
+      const hasDuplicate = _.flow([_.contains(id)])(favorites);

```

Comment:
Flow for one function? Realy? 😄 I think just `contains` will work.

Also, you can remove this var and just right:

`if (_.contains(id, favourites))`